project.md

```diff
@@ -11,41 +11,59 @@ _What need is met by your idea? Who are you helping? What is that person's
+
 ### Interface (syntax)
 
 _How might the user interact with the language? What does programming look
 like? Why is this the right way to interact with the problem domain?_
 
+A user would interact with this language in a similar format to what is the standard of Google/Excel sheets, which is like calling functions with multiple arguments in general purpose programming languages (i.e formula(arg1, arg2, ...) ). This is the right way to interact with the problem domain because many of the users are likely not coming from programming backgrounds, or at least this is not an expected requisite, so in asking them to write code, I am hoping to keep their experience as similar to that which they are used to in used the existing software, avoiding many of the technical concepts we use as programmers.
+
 ### Operation (semantics)
 
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When a program runs, it will check for function calls and their respective arguments, telling users if they have provided invalid arguments (i.e. argument which must be positive was given a negative value, 2 arguments given when 1 was expected, etc.) This would also be useful for users because in my experience, the error handling by software like Google and Excel is minimal and not very useful for determining what was wrong with the input. This will require more testing to ensure that the errors provided are more clear for the users who are not expected to have backgrounds in coding. Some examples of error handling I have considered before are: "division by zero error", "Invalid value for X", and "Not enough arguments. This function expects the following arguments: X, Y, and Z".
+
 ### Expressiveness
 
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It should be easy to store quantities and perform basic Economics formulas in google sheets such as: SUM() and VARIANCE(). This also means it should be possible to perform basic arithmetic operations (+, -, *, /) with the outputs of these formulas. People should be limited in what they can express to the scope of these operations and formulas from Google/Excel so that they are not confused when they combine different operations and see unclear behavior as they would not have the technical background to know how to resolve it or even recognize it as a bug at times.
+
+I am not fully certain of which finance formulas I want to implement, since I know I will have to limit them as there are many but I want to prioritize some of the most used ones such as: PV(), FV(), and NPV(). This will be an ongoing decision and depending on how much language design will go into each formula, I may do more or less to ensure I am not spending too much time per task relative to the functionality being gained.
+
```

Comment:
As we discussed in class I think it will be key for you to figure out which mathematical formulas you want to incorporate into your language. This will help you to narrow down your language choice and to help you with the overall direction of your project. One suggestion might to be to list out each possible function you are thinking about and then to do some kind of map from front end to back end of your program for each. this will help you to decided which ones will be best or most important for implementation.